pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMetadataProvider.java

```diff
@@ -43,6 +45,18 @@ public interface StreamMetadataProvider extends Closeable {
    */
   int fetchPartitionCount(long timeoutMillis);
 
+  /**
+   * Fetches the partition ids for a topic given the stream configs.
+   */
+  default Set<Integer> fetchPartitionIds(long timeoutMillis) {
```

Comment:
I think the default unfortunately needs to be `fetchStreamPartitionOffset` and get the partition ids from there. That's effectively what the previous behavior was, so anyone else like us who wasn't depending on sequentially increasing partition IDs will break

Reply:
Good point! Just realized that we can directly read the custom partition ids in `KafkaStreamMetadataProvider`. Applied the changes

Reply:
should we still use `fetchStreamPartitionOffset` as the default? this will break anyone upgrading with custom stream ingestion unless they can perfectly coordinate plugin deploys with new pinot deploys.

in our case, we could do it. But plugins live in a separate repo, so it would take a fair bit of coordination to get the latest libraries, update the plugin, deploy both server and plugin, and make sure nothing restarts in the middle

Reply:
By default this method is unsupported and will fall back to `computePartitionGroupMetadata()` at the caller side. We cannot directly use `computePartitionGroupMetadata()` as the default here because the arguments are different, and we don't want to force the caller to read the consumption status for all existing partitions

Reply:
oh good point. then yes, this lgtm